lib/phlex/fifo_cache_store.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Phlex::FIFOCacheStore
+	def initialize(max_bytesize: 2 ** 20)
+		@fifo = Phlex::FIFO.new(
+			max_bytesize:,
+			max_value_bytesize: max_bytesize
+		)
+	end
+
+	def fetch(key)
+		fifo = @fifo
+		key = map_key(key)
+
+		if (result = fifo[key])
+			result
+		else
+			result = yield
+			fifo[key] = result
+			result
+		end
+	end
+
+	def map_key(value)
```

Comment:
I wonder if this should keep track of the rough bitesize of the full key and raise if it gets too large, since the objects will never be GC’d while they’re in the cache. 🤔